web3/_utils/module_testing/shh_module.py

```diff
@@ -0,0 +1,389 @@
+        except:
+            assert True
+
+    #
+    # shh_key_pair
+    #
+    def test_shh_asymmetric_key_pair(self, web3):
+        # Test generating key
+        key_id = web3.geth.shh.newKeyPair()
+        assert web3.geth.shh.hasKeyPair(key_id)
+        assert len(web3.geth.shh.getPublicKey(key_id)) == 132
+
+        private_key = web3.geth.shh.getPrivateKey(key_id)
+        assert len(private_key) == 66
+        assert web3.geth.shh.deleteKeyPair(key_id)
+
+        # Test adding a key
+        assert not web3.geth.shh.hasKeyPair(key_id)
+        key_id = web3.geth.shh.addPrivateKey(private_key)
+        assert web3.geth.shh.hasKeyPair(key_id)
+        assert web3.geth.shh.deleteKeyPair(key_id)
+
+    def test_shh_symmetric_key_pair(self, web3):
+        # Test generating key
+        key_id = web3.geth.shh.newSymKey()
+        assert web3.geth.shh.hasSymKey(key_id)
+
+        key = web3.geth.shh.getSymKey(key_id)
+        assert len(key) == 66
+        assert web3.geth.shh.deleteSymKey(key_id)
+
+        # Test adding a key
+        assert not web3.geth.shh.hasSymKey(key_id)
+        key_id = web3.geth.shh.addSymKey(key)
+        assert web3.geth.shh.hasSymKey(key_id)
+        assert web3.geth.shh.deleteSymKey(key_id)
+
+    def test_shh_symmetric_key_pair_from_password(self, web3):
+        key_id = web3.geth.shh.generateSymKeyFromPassword('shh be quiet')
+
+        assert web3.geth.shh.hasSymKey(key_id)
+        assert len(web3.geth.shh.getSymKey(key_id)) == 66
+        assert web3.geth.shh.deleteSymKey(key_id)
+
+    #
+    # shh_post
+    #
+    def test_shh_post(self, web3):
+        receiver_pub = web3.geth.shh.getPublicKey(web3.geth.shh.newKeyPair())
+        assert web3.geth.shh.post({
+            "topic": "0x12345678",
+            "powTarget": 2.5,
+            "powTime": 2,
+            "payload": web3.toHex(text="testing shh on web3.py"),
+            "pubKey": receiver_pub,
+        })
+
+    #
+    # shh_properties
+    #
+    def test_shh_version(self, web3):
+        version = web3.geth.shh.version()
+        if '1.7' in web3.clientVersion:
+            assert version == '5.0'
+        else:
+            assert version == '6.0'
+
+    def test_shh_info(self, web3):
+        web3.geth.shh.setMaxMessageSize(1024)
+        web3.geth.shh.setMinPoW(0.5)
+
+        info = web3.geth.shh.info()
+
+        assert len(info) == 4
+        assert info["maxMessageSize"] == 1024
+        assert info["minPow"] == 0.5
+
+
+class ParityShhModuleTest():
+    #
+    # shh_filter
+    #
+    def test_shh_sync_filter(self, web3):
+        sender = web3.parity.shh.newKeyPair()
+        sender_pub = web3.parity.shh.getPublicKey(sender)
+
+        receiver = web3.parity.shh.newKeyPair()
+        receiver_pub = web3.parity.shh.getPublicKey(receiver)
+
+        topic = '0x13370000'
+        payloads = [web3.toHex(text="test message :)"), web3.toHex(text="2nd test message")]
+
+        shh_filter_id = web3.parity.shh.newMessageFilter({
```

Comment:
There seems to be a *lot* of similarity between these two implementations (geth and parity) of the tests.  Thoughts on trying to collapse these into something a little more DRY?

Reply:
I'd like to, but imo they aren't as similar as they may appear. Besides the different shh modules being tested against, the way parity implements `post`, `newMessageFilter`, `info` and `version` all differ in the params accepted by the jsonrpc endpoint, and the type of data returned. The only tests that are the same are `test_shh_asymmetric_key_pair` and `test_shh_symmetric_key_pair`. Beyond that it seems to me like sharing tests might require a fair amount of config overhead that might end up more confusing